### PR TITLE
Added COVID-19 Field to UntimelyExemptionModal

### PR DESCRIPTION
**Pull request by jcq**
Connects #13809

### Description
This adds a field to the form in `UntimelyExemptionModal` that specifies that the exemption is related to COVID-19

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Field added to `UntimelyExemptionModal` component
- [ ] Code is behind `covid_timeliness_exemption` feature toggle

### CODE REVIEW

**Comment by jcq** [comment]
This ended up including the code from @nanotone's PR (#13814), so we can likely close that when this gets completed.

**Comment by jcq** [comment]
## Screenshot of Updated View
Screen Shot 2020-03-31 at 10 54 47

**Comment by leikkisa** [comment]
Updated design selected by AMO:
COVID - Checkbox

File: client/app/intake/actions/decisionReview.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { ACTIONS, ENDPOINT_NAMES } from '../constants';
 import ApiUtil from '../../util/ApiUtil';
-import { analyticsCallback, submitIntakeCompleteRequest, submitIntakeReviewRequest } from './intake';
```
Comment (leikkisa) [inline client/app/intake/actions/decisionReview.js]: Surprised this hasn't been caught by the linter in so long!

File: client/app/intake/components/UntimelyExemptionModal.jsx [hunk 2]
```diff
@@ -1,111 +1,132 @@
...
-            name="untimely-exemption"
-            label="Did the applicant request an extension to the date requirements?"
-            strongLabel
-            vertical
-            options={BOOLEAN_RADIO_OPTIONS}
-            onChange={this.radioOnChange}
-            value={this.state.untimelyExemption === null ? null : this.state.untimelyExemption.toString()}
-          />
+      !state.untimelyExemption ||
+      (covidTimelinessExemption && state.untimelyExemption === 'true' && !state.untimelyExemptionCovid)
```
Comment (leikkisa) [inline client/app/intake/components/UntimelyExemptionModal.jsx]: With the checkbox, I'm okay with the value being nil if it's not checked. No strong opinion.

File: client/app/intake/components/UntimelyExemptionModal.jsx [hunk 3]
```diff
@@ -1,111 +1,132 @@
...
+          strongLabel
+          vertical
+          options={BOOLEAN_RADIO_OPTIONS}
+          onChange={(val) => setState({ ...state, untimelyExemption: val })}
+          value={state.untimelyExemption === null ? null : state.untimelyExemption.toString()}
+        />
+
+        {state.untimelyExemption === 'true' && (
+          <>
+            {covidTimelinessExemption && formType === 'higher_level_review' && (
```
Comment (leikkisa) [inline client/app/intake/components/UntimelyExemptionModal.jsx]: Per the update from AMO, we'll use a checkbox instead. Also, you can remove the check for higher level review, this is due to a change in the original AC. It'll be available for appeals... and in the next round of work for supplemental claims.

File: client/app/intake/components/UntimelyExemptionModal.jsx [hunk 4]
```diff
@@ -1,111 +1,132 @@
...
-
-    if (this.props.onSkip) {
-      btns.push({
-        classNames: ['usa-button', 'usa-button-secondary', 'no-matching-issues'],
-        name: this.props.skipText,
-        onClick: this.props.onSkip
-      });
+import { useSelector } from 'react-redux';
+
+const generateButtons = ({ cancelText, onCancel, onSubmit, submitText, skipText, onSkip, state, isInvalid }) => {
```
Comment (leikkisa) [inline client/app/intake/components/UntimelyExemptionModal.jsx]: Thank you as always for the code improvements!

File: spec/feature/intake/add_issues_spec.rb [hunk 5]
```diff
@@ -198,6 +198,49 @@
...
+          click_on "Establish EP"
+          expect(page).to have_content("Intake completed")
+
+          expect(RequestIssue.all.size).to eq(1)
+          untimely_issue = RequestIssue.first
+          expect(untimely_issue.covid_timeliness_exempt).to eq(true)
+        end
+      end
+
+      context "for appeal" do
```
[the remark this answers is not in the document]
Comment (jcq) [reply spec/feature/intake/add_issues_spec.rb]: Oh! I didn't realize that had been removed. I'll update.

File: client/app/intake/components/UntimelyExemptionModal.jsx (hunk 2 above)
Comment (jcq) [reply client/app/intake/components/UntimelyExemptionModal.jsx]: Yeah, I'm going to simplify this, given the change to checkbox

File: client/app/intake/components/UntimelyExemptionModal.jsx (hunk 3 above)
Comment (jcq) [reply client/app/intake/components/UntimelyExemptionModal.jsx]: This simplifies things anyway, so I'm happy to change

**Comment by jcq** [comment]
> Hi JC - this is looking nice! It isn't working on the edit page. Feature Toggles might be a good place to start looking because I didn't see any changes to code for what shows up on the edit page vs. not.

@leikkisa Ah, you're right — we need to add in the feature toggles for the edit view files. Am I understanding correctly that it's just for appeal & higher level reviews but not for supplemental claims?

**Comment by jcq** [comment]
## Final Screenshot
Screen Shot 2020-03-31 at 10 54 47

**Pull request merged**